Shell/AST.cpp

```diff
@@ -1116,8 +1116,8 @@ RefPtr<Value> Pipe::run(RefPtr<Shell> shell)
 
     auto pipe_write_end = new FdRedirection(STDIN_FILENO, -1, Rewiring::Close::Destination);
     auto pipe_read_end = new FdRedirection(STDOUT_FILENO, -1, pipe_write_end, Rewiring::Close::RefreshDestination);
-    first_in_right.redirections.append(*pipe_write_end);
-    last_in_left.redirections.append(*pipe_read_end);
+    first_in_right.redirections.empend(adopt(static_cast<Redirection&>(*pipe_write_end)));
+    last_in_left.redirections.empend(adopt(static_cast<Redirection&>(*pipe_read_end)));
```

Comment:
Please wrap the `new` in `adopt` directly instead of doing it later.

Reply:
With the double-`RefCounted` weirdness now fixed, this `static_cast` isn't needed anymore.